modules/calendar/app/services/calendar/create_ical_service.rb

```diff
@@ -0,0 +1,211 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require 'icalendar'
+require 'rails-html-sanitizer'
+
+module Calendar
+  class CreateICalService < ::BaseServices::BaseCallable
+    include TextFormattingHelper
+
+    def perform(work_packages:, calendar_name: "OpenProject Calendar")
+      ical_string = create_ical_string(work_packages, calendar_name)
+
+      ServiceResult.success(result: ical_string)
+    end
+
+    protected
+
+    def create_ical_string(work_packages, calendar_name)
+      calendar = Icalendar::Calendar.new
+
+      calendar.prodid = "-//OpenProject GmbH//OpenProject Core Project//EN"
+      calendar.x_wr_calname = calendar_name
+
+      work_packages.each do |work_package|
+        event = create_event(work_package)
+        event = add_attendee_value(event, work_package)
+
+        calendar.add_event(event)
+      end
+
+      calendar.to_ical
+    end
+
+    def create_event(work_package)
+      event = Icalendar::Event.new
+      event.uid = event_uid_value(work_package)
+      event.organizer = organizer_value(work_package)
+      event.summary = summary_value(work_package)
+      event.dtstart = dtstart_value(work_package)
+      event.dtend = dtend_value(work_package)
+      event.location = location_value(work_package)
+      event.description = description_value(work_package)
+
+      event
+    end
+
+    def event_uid_value(work_package)
+      "#{work_package.id}@#{host}"
+    end
+
+    def attendee_value(work_package)
+      [work_package.assigned_to&.name]
```

Comment:
@ulferts Interesting! Providing just the name has not caused any issues in multiple tested calendar applications, but yes, you're right, it not follows the iCalendar specification. But doing just this `["CN=#{work_package.assigned_to.name}"]` does not follow the specification either - as far as I understood. It might actually cause more errors in calendar applications because the format implies a contact link (e.g. mailto:)

I don't see a reason why exposing the assigned_to and author email addresses like `Icalendar::Values::CalAddress.new("mailto:#{work_package.author&.mail}", cn: work_package.author&.name)` which leads to `ORGANIZER;CN=Bob Bobbit:mailto:bobmail5.bobbit@bob.com` is not desired. The access to the iCalendar data is restricted and even more sensitive data (work_package information) are exposed here. So from a data-safety perspective I think it is OK.

Most calendar applications will probably create clickable links for the Author/Assignee names which enables the user to quickly compose an email to these people. This can be convenient on the one hand. On the other hand it simplifies communication bypassing OpenProject's comments which might harm communication and decision transparency.

Linking to the profile of the Authot/Assignee via web-link as an alternative of having a mailto value, seems not to follow the specification as well.

This needs to be discussed with the product team @marcalcobe @lindenthal